src/Api/Management/Entity/App.php

```diff
@@ -216,4 +219,20 @@ public function setScopes(string ...$scopes): void
     {
         $this->privateSetScopes(...$scopes);
     }
+
+    /**
+     * @param array $apiProducts
+     */
+    public function getApiProducts(): array
+    {
+        return $this->initialApiProducts;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setInitialApiProducts(array $initialApiProducts): void
```

Comment:
What should/would happen if this is called on an existing entity before a PUT?

Reply:
This should be called only on the creation of the App. 
https://github.com/apigee/apigee-edge-drupal/blob/3e5dcb11906aa49b3499098208c564ab5dedfcd1/src/Entity/Form/AppForm.php#L129

Reply:
Should be, but technically speaking nothing prevents it to be called on an existing app by a developer. Unless you throw an exception here... but this desing goes already againts interface the interface separation principle.

Being that said, I have no better idea than throwing an exception. Everything else would come with bigger refactoring - which I am unsure how long we can avoid...